# Exploration 2
Part of exploration initial exploration process

In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("../data/tax_data_2020.csv")
print(df.shape)
df.head(10)

(166452, 165)


STATEFIPS STATE  zipcode  agi_stub        N1     mars1     MARS2     MARS4  \
0          1    AL        0         1  785000.0  519980.0   85690.0  165290.0   
1          1    AL        0         2  554310.0  270870.0  121420.0  146470.0   
2          1    AL        0         3  290630.0  113280.0  124770.0   44570.0   
3          1    AL        0         4  181010.0   42010.0  120820.0   14410.0   
4          1    AL        0         5  269080.0   31310.0  224330.0    8270.0   
5          1    AL        0         6   81340.0    5000.0   72270.0     630.0   
6          1    AL    35004         1    1460.0    1010.0     200.0     210.0   
7          1    AL    35004         2    1400.0     760.0     280.0     310.0   
8          1    AL    35004         3    1020.0     460.0     350.0     190.0   
9          1    AL    35004         4     630.0     140.0     420.0      60.0   

        ELF    CPREP  ...   N85300    A85300    N11901     A11901    N11900  \
0  724170.0  22560.0  ...      0.0       0.0   57720.0    46577.0  674840.0   
1  515150.0  13260.0  ...      0.0       0.0   81770.0   112540.0  470410.0   
2  269700.0   6420.0  ...      0.0       0.0   70360.0   144380.0  220710.0   
3  168830.0   2570.0  ...      0.0       0.0   49500.0   135429.0  130670.0   
4  252360.0   3250.0  ...    100.0      20.0  103250.0   470206.0  165650.0   
5   76810.0    230.0  ...  41750.0  164094.0   43960.0  1216690.0   34560.0   
6    1340.0     40.0  ...      0.0       0.0     110.0       80.0    1210.0   
7    1310.0     40.0  ...      0.0       0.0     190.0      194.0    1210.0   
8     970.0     30.0  ...      0.0       0.0     230.0      357.0     800.0   
9     590.0      0.0  ...      0.0       0.0     180.0      386.0     450.0   

      A11900    N11902     A11902   N12000    A12000  
0  1827202.0  672200.0  1818867.0   2900.0    6089.0  
1  1445383.0  466960.0  1432458.0   4660.0   11648.0  
2   626662.0  216530.0   610170.0   5760.0   16235.0  
3   437179.0  126790.0   419324.0   3730.0   14903.0  
4   724529.0  156910.0   642895.0  11280.0   80064.0  
5   735671.0   24770.0   285560.0   9630.0  391783.0  
6     2807.0    1210.0     2795.0      0.0       0.0  
7     3378.0    1200.0     3335.0     30.0      62.0  
8     2131.0     790.0     2119.0      0.0       0.0  
9     1282.0     440.0     1276.0      0.0       0.0  

[10 rows x 165 columns]

In [3]:
df.A00100.mean()

152231.79502799606

### Check how many rows have a zipcode of 0, should probably drop these columns

In [4]:
df[df.zipcode == 0].shape

(306, 165)

306 rows with a zipcode of 0

Not sure what we could do with these rows

In [5]:
df.STATE.nunique()

51

In [6]:
df.STATE.value_counts().reset_index()

index  STATE
0     TX   9702
1     NY   9234
2     CA   8898
3     PA   8219
4     IL   7379
5     OH   5988
6     FL   5526
7     MI   5352
8     MO   5351
9     IA   4950
10    VA   4758
11    MN   4746
12    NC   4350
13    WI   4278
14    IN   4061
15    GA   3996
16    KY   3917
17    TN   3552
18    KS   3498
19    AL   3474
20    NJ   3282
21    OK   3282
22    WV   3054
23    WA   3006
24    AR   2934
25    MA   2892
26    NE   2862
27    LA   2688
28    MD   2430
29    CO   2381
30    SC   2243
31    ME   2220
32    MS   2219
33    OR   2148
34    AZ   1775
35    SD   1691
36    ND   1632
37    CT   1584
38    VT   1422
39    NH   1386
40    MT   1344
41    ID   1296
42    NM   1271
43    UT   1145
44    NV    786
45    WY    642
46    RI    426
47    HI    360
48    DE    342
49    AK    336
50    DC    144

looks like there is 6 rows with zipcode 0 for each state code (50 states + 1 for DC), one row for each AGI_STUB value

my guess would be these represent rows of txes filed by people where location was not known

### From this I think a good place to start could be grouping by State and AGI_STUB

In [7]:
gp = df.groupby(['STATE','agi_stub'])
gp.mean().round(2)

STATEFIPS   zipcode       N1    mars1    MARS2   MARS4  \
STATE agi_stub                                                           
AK    1               2.0  97866.27  3629.11  2899.46   287.68  351.79   
      2               2.0  97866.27  2866.43  1771.43   512.50  486.61   
      3               2.0  97866.27  1839.11   910.36   620.89  246.43   
      4               2.0  97866.27  1286.25   458.93   680.00  115.54   
      5               2.0  97866.27  2176.96   394.11  1655.00  100.36   
...                   ...       ...      ...      ...      ...     ...   
WY    2              56.0  82000.74  1202.24   708.04   279.53  185.23   
      3              56.0  82000.74   793.74   343.08   355.05   78.69   
      4              56.0  82000.74   552.90   140.47   369.81   31.68   
      5              56.0  82000.74   853.18   103.93   716.82   20.37   
      6              56.0  82000.74   239.07    29.25   198.22    0.56   

                    ELF   CPREP     PREP  DIR_DEP  ...  N85300   A85300  \
STATE agi_stub                                     ...                    
AK    1         3248.04  171.43  1270.71  2469.46  ...    0.00     0.00   
      2         2672.50   77.14  1006.61  2308.57  ...    0.00     0.00   
      3         1705.00   50.00   742.86  1376.25  ...    0.00     0.00   
      4         1190.00   27.68   565.00   893.04  ...    0.00     0.00   
      5         2015.89   51.61  1044.29  1290.00  ...    0.00     0.00   
...                 ...     ...      ...      ...  ...     ...      ...   
WY    2         1139.35   19.35   505.23   942.71  ...    0.00     0.00   
      3          748.60   17.01   391.78   573.55  ...    0.00     0.00   
      4          523.08    6.64   286.07   373.83  ...    0.00     0.00   
      5          811.21    9.25   490.19   485.42  ...    0.00     0.00   
      6          228.22    0.00   191.78    54.77  ...  138.79  2291.46   

                N11901   A11901   N11900   A11900   N11902   A11902  N12000  \
STATE agi_stub                                                                
AK    1         364.29   225.89  2846.79  5804.32  2830.18  5743.46   17.32   
      2         377.50   559.36  2474.82  7202.71  2453.21  7093.50   25.71   
      3         353.93   773.32  1481.07  4657.18  1455.71  4506.46   38.04   
      4         309.64   871.89   968.93  3540.75   943.75  3402.54   25.00   
      5         733.75  3373.57  1434.64  6839.11  1365.36  6188.86   93.21   
...                ...      ...      ...      ...      ...      ...     ...   
WY    2         158.13   224.67  1034.95  2855.87  1021.40  2795.08   18.60   
      3         168.04   356.13   632.52  1905.03   614.58  1815.10   21.68   
      4         127.85   331.21   415.05  1463.01   401.78  1397.33   17.57   
      5         291.21  1458.95   556.36  2720.79   515.79  2344.64   57.57   
      6         115.33  4888.34   115.51  5976.71    62.62  1228.69   54.86   

                 A12000  
STATE agi_stub           
AK    1           36.07  
      2           69.75  
      3          138.89  
      4           94.39  
      5          640.82  
...                 ...  
WY    2           52.47  
      3           65.14  
      4           74.17  
      5          414.65  
      6         4312.90  

[306 rows x 163 columns]

In [8]:
df.describe()

STATEFIPS        zipcode       agi_stub            N1  \
count  166452.000000  166452.000000  166452.000000  1.664520e+05   
mean       29.661128   48879.018474       3.499928  1.897116e+03   
std        15.122133   27173.175747       1.707865  3.803230e+04   
min         1.000000       0.000000       1.000000  0.000000e+00   
25%        18.000000   27023.000000       2.000000  7.000000e+01   
50%        29.000000   48855.000000       3.000000  2.700000e+02   
75%        42.000000   70661.000000       5.000000  1.110000e+03   
max        56.000000   99999.000000       6.000000  5.674590e+06   

              mars1         MARS2          MARS4           ELF          CPREP  \
count  1.664520e+05  1.664520e+05  166452.000000  1.664520e+05  166452.000000   
mean   9.527830e+02  6.486742e+02     251.089984  1.759313e+03      55.531685   
std    2.328303e+04  1.200091e+04    6098.937311  3.501830e+04    1453.363460   
min    0.000000e+00  0.000000e+00       0.000000  0.000000e+00       0.000000   
25%    0.000000e+00  4.000000e+01       0.000000  7.000000e+01       0.000000   
50%    8.000000e+01  1.100000e+02      30.000000  2.500000e+02       0.000000   
75%    4.100000e+02  3.800000e+02     100.000000  1.030000e+03      30.000000   
max    4.315530e+06  1.784370e+06  948660.000000  5.116340e+06  272920.000000   

               PREP  ...         N85300        A85300        N11901  \
count  1.664520e+05  ...  166452.000000  1.664520e+05  1.664520e+05   
mean   9.957618e+02  ...      65.666799  3.743286e+02  3.898242e+02   
std    2.047751e+04  ...    3351.833987  2.230338e+04  7.616816e+03   
min    0.000000e+00  ...       0.000000  0.000000e+00  0.000000e+00   
25%    5.000000e+01  ...       0.000000  0.000000e+00  2.000000e+01   
50%    1.500000e+02  ...       0.000000  0.000000e+00  6.000000e+01   
75%    5.800000e+02  ...       0.000000  0.000000e+00  2.300000e+02   
max    3.287230e+06  ...  991480.000000  6.385520e+06  1.125880e+06   

             A11901        N11900        A11900        N11902        A11902  \
count  1.664520e+05  1.664520e+05  1.664520e+05  1.664520e+05  1.664520e+05   
mean   2.505754e+03  1.451128e+03  5.318271e+03  1.412718e+03  4.313097e+03   
std    9.224512e+04  3.010481e+04  1.078037e+05  2.970901e+04  8.354194e+04   
min    0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
25%    2.000000e+01  5.000000e+01  1.860000e+02  5.000000e+01  1.650000e+02   
50%    1.450000e+02  2.000000e+02  7.180000e+02  1.900000e+02  6.320000e+02   
75%    6.560000e+02  8.000000e+02  2.836250e+03  7.600000e+02  2.504000e+03   
max    2.689319e+07  4.395870e+06  1.719525e+07  4.363130e+06  1.043208e+07   

              N12000        A12000  
count  166452.000000  1.664520e+05  
mean       48.894336  9.497740e+02  
std      1065.687240  4.533338e+04  
min         0.000000  0.000000e+00  
25%         0.000000  0.000000e+00  
50%         0.000000  0.000000e+00  
75%        30.000000  9.000000e+01  
max    211500.000000  1.124986e+07  

[8 rows x 164 columns]

List of columns I think could potentially be interesting to look at:
    
    A00100 - adjust gross income
    A02650 - total income amount
    A18300 - total taxes paid amount
    A11901 - tax due at time of filing amount
    A19900 - total overpayments amount

## Initial investigation of flat income tax

In [9]:
total_income_amount = df.A02650.sum()
total_income_amount

25622193370.060104

In [10]:
total_tax_due = df.A11901.sum()
total_tax_due

417087806.0

In [11]:
(total_tax_due / total_income_amount)*100

1.6278380229827356

In [12]:
(df.A02650*.016278).sum()

417078063.6778383

Is this saying a flat income tax of 1.627% would result in the same amount being collected from taxes? 

that does not seem to make much sense to me

In [13]:
df.A11901

0          46577.0
1         112540.0
2         144380.0
3         135429.0
4         470206.0
            ...   
166447       590.0
166448       947.0
166449      1052.0
166450      3897.0
166451     12002.0
Name: A11901, Length: 166452, dtype: float64

In [14]:
df.A02650

0         10172698.0
1         20123988.0
2         18005183.0
3         15830735.0
4         37080401.0
             ...    
166447      128723.0
166448      146431.0
166449      148463.0
166450      346623.0
166451      520735.0
Name: A02650, Length: 166452, dtype: float64

In [15]:
djjis

NameError: name 'djjis' is not defined